pp.c

```diff
@@ -6045,11 +6045,10 @@ PP(pp_split)
 	}
 	else {
 	    if (!AvREAL(ary)) {
-		I32 i;
 		AvREAL_on(ary);
 		AvREIFY_off(ary);
-		for (i = AvFILLp(ary); i >= 0; i--)
-		    AvARRAY(ary)[i] = &PL_sv_undef; /* don't free mere refs */
+		/* don't free mere refs */
+		Zero(AvARRAY(ary),AvFILLp(ary) + 1,SV*);
```

Comment:
I wonder if you need a protection on `AvFILLp(ary) >= 0`, I can find some code setting `AvFILLp(av) = -1` for example

Reply:
Hmmm, I had thought it was safe to do without, but after having a quick skim I suddenly wonder if this is the correct commit at all:
* Pretty much immediately prior to this branch, _pp_split_ calls _av_clear(ary)_
* _av_clear()_ sets `AvFILLp(av) = -1;` - which seems to make the existing `for` loop redundant.

I'll have a proper look later, but if anyone has time/motivation to double-check that thinking, I'd be grateful.

Reply:
I would probably preserve the Zero here to preserve the existing behavior, and do the investigation you are talking in a follow up commit, as this seems a potential (or not) optimization.

Reply:
After checking the `AvFILLp(ary)` protection is not needed.
`Zero` is just a wrapper around `memzero` which is a wrapper around `memset` and memset is protected... with a check > 0` so this is good to merge like it